wcfsetup/install/files/lib/form/EmailActivationForm.class.php

```diff
@@ -97,8 +97,12 @@ public function save() {
 			'newEmail' => '',
 			'reactivationCode' => 0
 		];
-		if ($this->user->activationCode != 0 && REGISTER_ACTIVATION_METHOD == 1) {
-			$data['activationCode'] = 0;
+		if (!$this->user->isEmailConfirmed() && (REGISTER_ACTIVATION_METHOD & User::REGISTER_ACTIVATION_USER)) {
+			$data['emailConfirmed'] = null;
+			
+			if (!(REGISTER_ACTIVATION_METHOD & User::REGISTER_ACTIVATION_ADMIN)) {
+				$data['activationCode'] = 0;
+			}
```

Comment:
Is this behavior correct? Consider the following situation:
- The user is not yet activated.
- The user changes the email address in the AccountManagement.
- The user's activationCode will be zeroed out.
- But `enable` is never called, thus not assigning the proper groups.

Instead `confirmEmail` should be invoked which does the right thing.

Reply:
It's the same as before. In other words: If it was broken before, it still is. I fix the problem here now and then later in 5.2 and 3.1.